### PR TITLE
Add information about the memory consumption

### PR DESCRIPTION
**Pull request by mikeSimonson**
of the archive cronjob.
This can be helpfull when you approach the memory limit.

### CODE REVIEW

**Comment by mikeSimonson** [comment]
The build is failing for reason that are unrelated to this PR.
All the tests that this PR broke are fixed.

File: core/CronArchive.php [hunk 1]
```diff
@@ -413,6 +414,7 @@ public function run()
             Piwik::postEvent('CronArchive.archiveSingleSite.start', array($idSite));
 
             $completed = $this->archiveSingleSite($idSite, $requestsBefore);
+            $this->logger->info("Mermory usage: " . $this->getHumanReadableBytesSize(memory_get_usage()));
```
Comment (mattab) [inline core/CronArchive.php]: typo should be `Memory`

File: core/CronArchive.php [hunk 2]
```diff
@@ -460,6 +462,12 @@ public function run()
         );
 
         $this->logger->info($timer->__toString());
+        $this->logger->info("Mermory peak usage: " . $this->getHumanReadableBytesSize(memory_get_peak_usage()));
```
Comment (mattab) [inline core/CronArchive.php]: typo should be `Memory`

File: core/CronArchive.php [hunk 3]
```diff
@@ -460,6 +462,12 @@ public function run()
...
+    }
+
+    private function getHumanReadableBytesSize($memoryInBytes, $precision = 2) {
```
Comment (mattab) [inline core/CronArchive.php]: Could you please format the code consistently with our code / PSR notation, ie add line breaks and spacing

**Comment by mattab** [comment]
Hi @mikeSimonson 
Thanks for the PR. Started reviewing it but then realised that this script is not actually the script which runs the archiving tasks. They are either
1. Run in a separate thread using our console command climulti:request (if supported)
2. Or they are run by calling the Piwik API over HTTP 

In both cases the memory of the archiving script will not raise much as the actual memory hungry processes are in separate threads.

Therefore I would say this PR is not needed. Could you clarify whether it was helpful for you at some point, and why?

**Comment by mattab** [comment]
@mikeSimonson Please see my last comment for questions. For now closing the PR but happy to re-open if it's actually useful :+1:

**Comment by mikeSimonson** [comment]
@mattab Sorry I thought I had already replied.

That commands is also directly called by cronjobs ( [from the docs](https://piwik.org/docs/setup-auto-archiving/#linuxunix-how-to-set-up-a-crontab-to-automatically-archive-the-reports) ).
It's in this case that it's useful for me as the log will give me more info as to why it might have failed.

**Comment by mattab** [comment]
@mikeSimonson would you mind sending here an example of the archive log output which shows the memory increase/decrease or another interesting use case? Thanks

**Comment by mikeSimonson** [comment]
```
INFO [2017-08-27 22:40:19] Archived website id = 3, period = year, 0 segments, 1495404 visits in last 2 years, 1495404 visits this year, Time elapsed: 45.997s
INFO [2017-08-27 22:40:19] Archived website id = 3, 4 API requests, Time elapsed: 35.564s [3/9 done]
INFO [2017-08-27 22:40:19] Mermory usage: 123.55MB
INFO [2017-08-27 22:40:19] - no new tracking data for website id 4 since 2017-08-27 22:35:11 UTC (since the last successful archiving)
INFO [2017-08-27 22:40:19] Skipped website id 4 as archiving is not needed
```

**Pull request closed without merging**